app/models/user.rb

```diff
@@ -641,25 +631,18 @@ def arel_select_species_list_ids_in_users_projects
   def interest_in(object)
     @interests ||= {}
     @interests["#{object.class.name} #{object.id}"] ||= begin
-      sql = interest_in_sql(object)
-      case Interest.connection.select_value(sql).to_s
-      when "1"
+      i = Interest.where(
+        user_id: id, target_type: object.class.name, target_id: object.id
+      ).limit(1).pluck(:state).first
+      case i
+      when true
         :watching
-      when "0"
+      when false
         :ignoring
       end
```

Comment:
Ternary operator would shorten this
```suggestion
      i ? :watching : :ignoring
```

Reply:
Ok. Just tried `i.to_bool` and `i.to_i` — that didn't cut it either. Merging it with this case statement!

Reply:
Thanks for the clarification., and my apologies:  I kicked off this whole thread by assuming it was a boolean and trying to use a ternary operator.